src/librustc/hir/lowering.rs

```diff
@@ -4058,16 +4058,16 @@ impl<'a> LoweringContext<'a> {
                 // expand <head>
                 let head = self.lower_expr(head);
                 let head_sp = head.span;
+                let desugared_span = self.allow_internal_unstable(
```

Comment:
Am I right that **desugared_span** is just the renaming of **next_sp** ?

Reply:
Yes, that's correct. I just moved it so that both the spans that were being referenced multiple times were in the same place.